ggml-backend.h

```diff
@@ -78,6 +81,15 @@ extern "C" {
     static inline void ggml_backend_graph_compute(struct ggml_backend * backend, struct ggml_cgraph * cgraph) { backend->interface->graph_compute(backend->context, cgraph); }
 
     // buffer and tensor allocation
+    // TODO:
+    //  - return "struct ggml_buffer *"
+    //  - fix namings:
+    //    - ggml_backend_alloc_buffer -> ggml_backend_buffer_alloc
+    //    - ggml_backend_free_buffer  -> ggml_backend_buffer_free
+    //    - ggml_backend_reset_buffer -> ggml_backend_buffer_reset
+    //    - ggml_backend_alloc_tensor -> ggml_backend_tensor_alloc
+    //    - ggml_backend_tensor_cpy   -> ggml_backend_tensor_copy
```

Comment:
I will rename these in my branch. The allocation interface is going to change significantly to be able to implement https://github.com/ggerganov/ggml/issues/288.

Reply:
Should we rename these too?
```
ggml_backend_set_tensor_async -> ggml_backend_tensor_set_async
ggml_backend_get_tensor_async -> ggml_backend_tensor_get_async
ggml_backend_set_tensor       -> ggml_backend_tensor_set
ggml_backend_get_tensor       -> ggml_backend_tensor_get
```

Reply:
>The allocation interface is going to change significantly

To clarify, the changes in the backend interface are minimal, just shuffling some code around, so this is nothing that you have to worry about when implementing the Metal backend. I am just going to add a `ggml_allocator` interface that will be able to handle more complex allocation than just increasing a pointer.